### ISSUE TITLE
Support for custom pretrained models

### ISSUE THREAD

**Issue by prinshul** [fcdl94/MiBv2#2]
Hi

Is it possible to use pretrained models other than trained by inplace_abn? Like SAM?

Can you please suggest how to use them?

**Comment by prinshul** [comment]
In short, is it mandatory to use Inplace_ABN pretrained models? Can we use other pretrained models trained in a different way or not use pretrained model at all?

**Comment by fcdl94** [comment]
The code has been designed to use the deeplab architecture. However, loss functions and the framework could also be employed using any other architecture, given it outputs a final class-probability mask (take a look at CoMFormer [here](https://github.com/fcdl94/CoMFormer), where I adapted MiB to Mask2Former).
Anyway, adapting the code to very different architectures (e.g. Mask2Former) may require substantial changes.

**Comment by fcdl94** [comment]
It is not mandatory to use Inplace_ABN models or use any pretraining (indeed, performance may be very different).
